samcli/lib/list/testable_resources/testable_resources_producer.py

```diff
@@ -0,0 +1,426 @@
+            if api_type == APIGatewayEnum.API_GATEWAY:
+                response = self.apigateway_client.get_stages(restApiId=api_id)
+                search_key = "item"
+                stage_name_key = "stageName"
+            elif api_type == APIGatewayEnum.API_GATEWAY_V2:
+                response = self.apigatewayv2_client.get_stages(ApiId=api_id)
+                search_key = "Items"
+                stage_name_key = "StageName"
+            if not response.get(search_key, []):
+                return response_list
+            for item in response.get(search_key, []):
+                if item.get(stage_name_key, None):
+                    response_list.append(item.get(stage_name_key, ""))
+            return response_list
+        except ClientError as e:
+            if get_client_error_code(e) == "NotFoundException":
+                return []
+            LOG.error("ClientError Exception : %s", str(e))
+            raise SamListUnknownClientError(msg=str(e)) from e
+        except BotoCoreError as e:
+            LOG.error("Botocore Exception : %s", str(e))
+            raise SamListUnknownBotoCoreError(msg=str(e)) from e
+
+    def build_api_gw_endpoints(self, physical_id: str, stages: list) -> list:
+        """
+        Builds the default api gateway endpoints
+
+        Parameters
+        ----------
+        physical_id: str
+            The physical ID of the api resource
+        stages: list
+            A list of stages for the api resource
+
+        Returns
+        -------
+        api_list: List[Any]
+            The list of default api gateway endpoints
+        """
+        api_list = []
+        for stage in stages:
+
+            api_list.append(f"https://{physical_id}.execute-api.{self.region}.amazonaws.com/{stage}")
+        return api_list
+
+    def get_cloud_testable_resources(self, stacks: list) -> list:
+        """
+        Gets a list of cloud testable resources
+
+        Parameters
+        ----------
+        stacks: list
+            A list containing the local stack
+
+        Returns
+        -------
+        testable_resources_list: List[Any]
+            A list of cloud testable resources
+        """
+        testable_resources_list = []
+        local_stack = stacks[0]
+        local_stack_resources = local_stack.resources
+        seen_testable_resources = set()
+        response = self.get_resources_info()
+        response_domain_dict = get_response_domain_dict(response)
+        custom_domain_substitute_dict = get_custom_domain_substitute_list(response, stacks, response_domain_dict)
+
+        for deployed_resource in response.get(STACK_RESOURCES, {}):
+            if deployed_resource.get(RESOURCE_TYPE, "") in TESTABLE_RESOURCE_TYPES:
+                endpoint_function_url: Any
+                paths_and_methods: Any
+                endpoint_function_url = "-"
+                paths_and_methods = "-"
+                if deployed_resource.get(RESOURCE_TYPE, "") == "AWS::Lambda::Function":
```

Comment:
There are too many if elses here. Can we split this function apart to do certain activities based on what resource type it is as we iterate?

Reply:
I've split the function